.github/workflows/build.yml

```
@@ -32,101 +32,91 @@ on:
-            # this is done to include the correct branch and git information inside the build
-            echo "docker_ref=${{ github.event.pull_request.head.ref }}" >> $GITHUB_OUTPUT
-          elif [[ "${{ github.event_name }}" == "push" ]]; then
-            echo "docker_ref=${{ github.ref_name }}" >> $GITHUB_OUTPUT
-          fi
-
-      - name: Compute Docker Tag
-        if: ${{ steps.set-docker-build.outputs.docker_build == 'true' }}
-        uses: actions/github-script@v7
-        id: compute-tag
+      - uses: actions/checkout@v4
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
         with:
-          result-encoding: string
-          script: |
-            if (context.eventName === "pull_request") {
-              return "pr-" + context.issue.number;
-            }
-            if (context.eventName === "release") {
-              return "latest";
-            }
-            if (context.eventName === "push") {
-              if (context.ref.startsWith("refs/tags/")) {
-                return context.ref.slice(10);
-              }
-              if (context.ref === "refs/heads/develop") {
-                return "develop";
-              }
-            }
-            return "FALSE";
-
-      - name: Set Docker Tag
-        id: set-docker-tag
-        run: |
-          if [[ ${{ steps.compute-tag.outputs.result  != 'FALSE' }} ]]; then
-            echo "docker_build_tag=${{ steps.compute-tag.outputs.result }}" >> $GITHUB_OUTPUT
-          fi
-
+          node-version: '${{ env.node }}'
+          cache: 'npm'
+      - name: Setup Java
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'temurin'
+          java-version: '${{ env.java }}'
+          cache: 'gradle'
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v3
+      - name: Production Build
+        run: ./gradlew -Pprod -Pwar clean bootWar
+      - name: Upload Artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: Artemis.war
+          path: build/libs/Artemis-*.war
+      - name: Upload Release Artifact
+        if: github.event_name == 'release' && github.event.action == 'created'
+        uses: actions/upload-release-asset@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          upload_url: ${{ github.event.release.upload_url }}
+          asset_path: build/libs/Artemis-${{ github.event.release.tag_name }}.war
+          asset_name: Artemis.war
+          asset_content_type: application/x-webarchive
 
-  call-build-workflow:
-    name: Call Build Workflow
-    needs: define-inputs
-    uses: ./.github/workflows/reusable-build.yml
+  docker:
+    name: Build and Push Docker Image
+    if: ${{ github.event_name != 'pull_request' || github.event.pull_request.head.repo.full_name == 'ls1intum/Artemis' }}
+    uses: ls1intum/.github/.github/workflows/build-and-push-docker-image.yml@main
```

As discussed in our meeting today - i released the new version for our github actions :)
```suggestion
    uses: ls1intum/.github/.github/workflows/build-and-push-docker-image.yml@v1.1.0
```